optimum/onnx/configuration.py

```diff
@@ -0,0 +1,229 @@
+
+import torch
+from transformers.onnx import OnnxConfig, OnnxConfigWithPast, OnnxSeq2SeqConfigWithPast
+
+
+logger = logging.get_logger(__name__)
+
+
+class OnnxConfigWithLoss(OnnxConfig, ABC):
+    """
+    Wrapper for the childern classes of `transformers.onnx.OnnxConfig` to export the model through the ONNX format with loss in outputs.
+    """
+
+    _tasks_to_extra_inputs = {
+        "default": OrderedDict({"labels": {0: "batch"}}),
+        "masked-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "causal-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "seq2seq-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "sequence-classification": OrderedDict({"labels": {0: "batch"}}),
+        "token-classification": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "multiple-choice": OrderedDict({"labels": {0: "batch"}}),
+        "question-answering": OrderedDict(
+            {
+                "start_positions": {0: "batch"},
+                "end_positions": {0: "batch"},
+            }
+        ),
+        "image-classification": OrderedDict({"labels": {0: "batch"}}),
+    }
+    _tasks_to_extra_outputs = {
+        "default": OrderedDict({"loss": {}}),
+    }
+
+    def __init__(self, config: OnnxConfig):
+        self.__dict__ = config.__dict__
+        self._onnx_config = config
+        if self.task not in self._tasks_to_extra_inputs:
+            raise ValueError(
+                f"{self.task} is not a supported task, supported tasks: {self._tasks_to_extra_inputs.keys()}"
+            )
+
+    @classmethod
+    def from_model_config(cls, config: OnnxConfig) -> "OnnxConfigWithLoss":
+        """
+        Instantiate a OnnxConfigWithLoss for a specific model
+        Args:
+            config: The model's configuration to use when exporting to ONNX
+        Returns:
+            OnnxConfigWithLoss for this model
+        """
+        return cls(config)
+
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        """
+        Mapping containing the axis definition of the input tensors(including labels) to provide to the model
+        Returns:
+            For each input: its name associated to the axes symbolic name and the axis position within the tensor
+        """
+        inputs = self._onnx_config.inputs
+        inputs.update(self._tasks_to_extra_inputs[self.task])
+        return inputs
+
+    @property
+    def outputs(self) -> Mapping[str, Mapping[int, str]]:
+        """
+        Mapping containing the axis definition of the output tensors(including loss) to provide to the model
+        Returns:
+            For each output: its name associated to the axes symbolic name and the axis position within the tensor
+        """
+        common_outputs = self._onnx_config.outputs
+        common_outputs.update(self._tasks_to_extra_outputs["default"])
+        if "loss" in common_outputs.keys():
```

Comment:
Yes, but I'm under the impression that you add the loss to `common_outputs` then check right after that it contains it?
I might be missing something here, just wanted to point it out.

Reply:
Oh I get what you mean now, sorry that I misunderstood it. Yeah in the case that we only have one `default` case, The if statement is not necessary. I've just put things this way in case we call loss differently in some cases. will replace "loss" by `common_outputs.keys` and delete the `if`